Added Declaring Variables

Jon2211 opened the pull request:
Added to beginner section: declaring an untyped, a typed and a dynamic variable

Gama11 commented on assets/content/cookbook/Beginner/declare-variables.md:
```diff
@@ -0,0 +1,34 @@
+
+# Declare variables
+
+Create a variable without type. Although the type is not explicity declared, you cannot reassign the variable later to a different type.
```
It's not really a "variable without a type", just without a type _hint_.

Jon2211 replied:
ok - I'll update

Jon2211 replied:
or maybe without a declared type

dstrekelj commented:
I feel this entry should also mention that Haxe allows declaration of multiple variables with the same identifier (or _redecleration_ of existing variables). I'd like to see it paired up with a short example showing the generated output, but that's just a preference.

This "issue" was raised on IRC not long ago, and it seems like a thing that might trip up newcomers. Having it in the cookbook would be smart.

Jon2211 commented:
1. Should this entry discuss base types or should that be another entry?
2.  Should this entry cover private/public/inline/static etc
3. access modifiers?

dstrekelj commented:
1. I'd save base types for their own entry. There's only three of them, but they'll take up about as much space as this entry does now. Besides, covering base types on their own would allow you to go into greater detail.

2. Same as above. They are good candidates for a standalone entry in the cookbook. Perhaps even more so than base types.

3. Do you mean mutators and accessors? If yes, then I say cover them with the _access modifiers_ from the previous point. If no, then I'm not sure what you mean.

dstrekelj commented:
Oh, and [here](https://groups.google.com/forum/#!topic/haxelang/Lqr7eVZEZgQ)'s another reason to cover variable redeclaration. Kind of funny this happened not long after I mentioned it here. "Speak of the devil" and all that.

markknol commented:
I think we should be carefu lwhen speaking about dynamic. I suggest to leave it out here. Haxe has a great type system, dynamic is a gateway to error phrone code. I rather avoid it or have a snippets/tutorial that explains why and how to avoid it.

Jon2211 commented:
Made some modifications to the entry.  Removed the dynamic type entry, added example of the null state of variables when created, and made an example of redeclaring variables in the local scope vs class scope.

Simn commented on assets/content/cookbook/Beginner/declare-variables.md:
```diff
@@ -0,0 +1,56 @@
+
+### Inital values of variables
+
+In Haxe, variables are null until used - including number types.
```
That is not true for static targets.

Simn commented:
This example conflates local variables and type inference in a way which I think causes more confusion than anything. Thank you for the contribution, but I prefer not to include this example for the time being.
